### PR TITLE
Add Xcode project and fix build

### PR DESCRIPTION
**Pull request by gshahbazian**
Add gitignore items for Xcode stuffs

### CODE REVIEW

File: Sources/http.swift [hunk 1]
```diff
@@ -20,16 +20,17 @@ public class HTTP {
 }
 
   init(port: UInt16) {
-    serverSocket = socket(AF_INET, Int32(SOCK_STREAM.rawValue), 0)
+    serverSocket = socket(AF_INET, Int32(SOCK_STREAM), 0)
```
Comment (huytd) [inline Sources/http.swift:26]: Have you tested this on Linux? Because when I tried to build on Linux, this line will caused some errors

Comment (gshahbazian) [inline Sources/http.swift:26]: Will try

Comment (huytd) [inline Sources/http.swift:26]: I think we should make another branch for Mac and another branch for Linux :D your changes work well on Mac. But Swift for Linux has some different :D

Comment (gshahbazian) [inline Sources/http.swift:26]: Fixed.

**Pull request merged**